### PR TITLE
Add docs for java and android sdks

### PR DESCRIPTION
**Pull request by gomathyk**

### CODE REVIEW

File: en/docs/fragments/register-a-spring-boot-app.md [hunk 1]
```diff
@@ -1,55 +1,24 @@
...
- 
- 7. Enter **Callback URL**. 
-      Add the flowwing two callBack URLs: 
+1.	Start WSO2 IS.
+
+2.	Access the Management Console (https://<IS_HOST\>:<PORT\>/carbon) to create a service provider.
+
+3.	Navigate to the **Service Providers** tab listed under the **Identity** section and click **Add**.
+
+4.	Provide the name for the service provider as `sample-app` and click **Register**.
```
Comment (dewniMW) [inline en/docs/fragments/register-a-spring-boot-app.md:7]: shall we use the register-a-service-provider.md fragment here?

File: en/docs/fragments/register-a-spring-boot-app.md [hunk 2]
```diff
@@ -1,55 +1,24 @@
...
-      Add the flowwing two callBack URLs: 
+1.	Start WSO2 IS.
+
+2.	Access the Management Console (https://<IS_HOST\>:<PORT\>/carbon) to create a service provider.
+
+3.	Navigate to the **Service Providers** tab listed under the **Identity** section and click **Add**.
+
+4.	Provide the name for the service provider as `sample-app` and click **Register**. 
+
+5.	Now you will be redirected to the **Edit Service Provider** page.
```
Comment (dewniMW) [inline en/docs/fragments/register-a-spring-boot-app.md:9]: also after registering the SP the user will directly land in the edit view of the new SP. hence this step can be removed IMO

File: en/docs/fragments/register-a-spring-boot-app.md [hunk 3]
```diff
@@ -1,55 +1,24 @@
...
+
+6.	Expand the  **Inbound Authentication Configuration** section and click **Configure** under the **OAuth/OpenID Connect Configuration** section.
+
+7.	Provide the callback URL as `http://localhost:8080/spring-boot-app/login/oauth2/code/wso2`.
+
+8.	Click **Update**.
+
+9.	Once the service provider is registered, you will be redirected to the **Service Provider Details** page. Here, expand the **Inbound Authentication 			Configuration** section and click the **OAuth/OpenID Connect Configuration** section. Copy the value of  `OAuth Client Key` shown here.
```
Comment (dewniMW) [inline en/docs/fragments/register-a-spring-boot-app.md:17]: `Inbound Authentication 			Configuration` shall we remove the unwanted spacing from here?

Comment (dewniMW) [inline en/docs/fragments/register-a-spring-boot-app.md:17]: `Once the service provider is registered,` part can be removed IMO

File: en/docs/fragments/register-a-spring-boot-app.md [hunk 4]
```diff
@@ -1,55 +1,24 @@
...
-                        
-
-**Eg:**
  
 | Field                 | Value                                                                                                 | 
 | --------------------- | -----------------------------                                                                         | 
 | Service Provider Name | sample-app                                                                                            |
 | Description           | This is a spring-boot application                                                                     | 
-| CallBack Url          | http://localhost:8080/sprinb-boot-app/login/oauth2/code/wso2, http://localhost:8080/spring-boot-app   |
+| CallBack Url          | http://localhost:8080/spring-boot-app/login/oauth2/code/wso2                                          |
```
Comment (dewniMW) [inline en/docs/fragments/register-a-spring-boot-app.md:24]: is it confirmed that `http://localhost:8080/spring-boot-app` is not needed?

File: en/docs/fragments/register-a-spring-boot-app.md [hunk 5]
```diff
@@ -1,55 +1,24 @@
...
+
+3.	Navigate to the **Service Providers** tab listed under the **Identity** section and click **Add**.
+
+4.	Provide the name for the service provider as `sample-app` and click **Register**. 
+
+5.	Now you will be redirected to the **Edit Service Provider** page.
+
+6.	Expand the  **Inbound Authentication Configuration** section and click **Configure** under the **OAuth/OpenID Connect Configuration** section.
+
+7.	Provide the callback URL as `http://localhost:8080/spring-boot-app/login/oauth2/code/wso2`.
```
Comment (dewniMW) [inline en/docs/fragments/register-a-spring-boot-app.md:13]: Any specific remove to not include the information available at https://github.com/wso2/docs-is/pull/2460/files#diff-68ed9a8b6ad3cdcce8550ce0e7bcbd73c1fee5cece64d643cd2ac39a14201270L23-L26 ?

File: en/docs/fragments/register-mobile-application.md [hunk 6]
```diff
@@ -1,34 +1,24 @@
...
- 
- 10. Click on the **Access** tab and note the **Client ID** that appears. 
\ No newline at end of file
+1.	Start WSO2 IS.
+2.	Access the Management Console (https://<IS_HOST\>:<PORT\>/carbon) to create a service provider.
+	![Management Console](../../../assets/img/fragments/android-sp.png)
+
+3.	Navigate to the **Service Providers** tab listed under the **Identity** section and click **Add**.
+
+4.	Provide the name for the service provider as `sample-app` and click **Register**.
```
Comment (dewniMW) [inline en/docs/fragments/register-mobile-application.md:7]: refer https://github.com/wso2/docs-is/pull/2460/files#r584441354

File: en/docs/quick-starts/android-sample.md [hunk 7]
```diff
@@ -1,105 +1,102 @@
...
+{!fragments/register-mobile-application.md!}
 
+---
 
-1. Download the [android sample](https://github.com/wso2-extensions/identity-samples-android/archive/master.zip) and
- Unzip it.
+## Set up the sample
 
+1.  Download the [android sample](https://github.com/wso2-extensions/identity-samples-android/archive/master.zip) and
+ Unzip it.
```
Comment (dewniMW) [inline en/docs/quick-starts/android-sample.md:18]:
```suggestion
 unzip it.
```

File: en/docs/quick-starts/android.md [hunk 8]
```diff
@@ -2,15 +2,21 @@
 
 !!! Tip 
     [Try Our Sample](../android-sample)
-     
+
+---
```
Comment (dewniMW) [inline en/docs/quick-starts/android.md:6]: why is this needed?

File: en/docs/fragments/register-a-spring-boot-app.md (hunk 1 above)
Comment (gomathyk) [reply en/docs/fragments/register-a-spring-boot-app.md:7]: We use different fragments for each app in samples since we need to give the sample values inline (For example, Provide the callback URL as `http://localhost:8080/spring-boot-app/login/oauth2/code/wso2`.)

File: en/docs/fragments/register-a-spring-boot-app.md (hunk 4 above)
Comment (gomathyk) [reply en/docs/fragments/register-a-spring-boot-app.md:24]: Yes, in 5.11.0 we have mentioned only one callback URL while providing instructions in terms of management console. https://is.docs.wso2.com/en/latest/develop/spring-boot/

File: en/docs/fragments/register-a-spring-boot-app.md (hunk 1 above)
Comment (dewniMW) [reply en/docs/fragments/register-a-spring-boot-app.md:7]: the aforementioned fragment doesn't contain info on callback URL

Comment (gomathyk) [reply en/docs/fragments/register-a-spring-boot-app.md:7]: Please refer step 7 and beyond of the register-a-spring-boot-app fragment.

File: en/docs/fragments/register-a-spring-boot-app.md (hunk 5 above)
Comment (gomathyk) [reply en/docs/fragments/register-a-spring-boot-app.md:13]: Added

File: en/docs/fragments/register-a-spring-boot-app.md (hunk 2 above)
Comment (gomathyk) [reply en/docs/fragments/register-a-spring-boot-app.md:9]: Removed

File: en/docs/fragments/register-a-spring-boot-app.md [hunk 9]
```diff
@@ -1,55 +1,32 @@
...
- 
- 7. Enter **Callback URL**. 
-      Add the flowwing two callBack URLs: 
+   !!!tip
+      The **Callback URL** is the exact location in the service provider's application where a authorization code
+      would be sent. This should be always `{baseUrl}/login/oauth2/code/wso2`.
+
+      Example: http://localhost:8080/login/oauth2/code/wso2
+
+      If you want configure oidc logout, then you need to add a post-logout-url as well.
```
Comment (dewniMW) [inline en/docs/fragments/register-a-spring-boot-app.md:21]: shall we also include this example: https://github.com/wso2/docs-is/pull/2460/files#diff-68ed9a8b6ad3cdcce8550ce0e7bcbd73c1fee5cece64d643cd2ac39a14201270L31 ?

File: en/docs/fragments/register-mobile-application.md [hunk 10]
```diff
@@ -1,34 +1,24 @@
...
\ No newline at end of file
+1.	Start WSO2 IS.
+2.	Access the Management Console (https://<IS_HOST\>:<PORT\>/carbon) to create a service provider.
+	![Management Console](../../../assets/img/fragments/android-sp.png)
+
+3.	Navigate to the **Service Providers** tab listed under the **Identity** section and click **Add**.
+
+4.	Provide the name for the service provider as `sample-app` and click **Register**. 
+
+5.	Now you will be redirected to the **Edit Service Provider** page.
```
Comment (dewniMW) [inline en/docs/fragments/register-mobile-application.md:9]: refer https://github.com/wso2/docs-is/pull/2460#discussion_r584441876

File: en/docs/fragments/register-mobile-application.md [hunk 11]
```diff
@@ -1,34 +1,24 @@
...
+6.	Expand the  **Inbound Authentication Configuration** section and click **Configure** under the **OAuth/OpenID Connect Configuration** section.
+
+7.	Provide the following values for the respective fields and click **Update** while keeping other default settings as it is.
+	-  Callback Url - wso2sample://oauth2
+	-  PKCE Mandatory - True
+	-  Allow authentication without the client secret - True
+
+8.	Click **Update**.
+
+9.	Once the service provider is registered, you will be redirected to the **Service Provider Details** page. Here, expand the **Inbound Authentication 			Configuration** section and click the **OAuth/OpenID Connect Configuration** section. Copy the value of  `OAuth Client Key` shown here.
```
Comment (dewniMW) [inline en/docs/fragments/register-mobile-application.md:20]: shall we remove the unwanted spacing

**Review by dewniMW — APPROVED** [review]
